book/src/development/infrastructure/sync.md

```diff
@@ -15,107 +15,82 @@ done in a bi-weekly basis if there's no urgent changes. This is done starting on
-> script and changing `sh` to `bash`.
-
-## Defining remotes
-
-You may want to define remotes, so you don't have to type out the remote
-addresses on every sync. You can do this with the following commands (these
-commands still have to be run inside the `rust` directory):
-
-```bash
-# Set clippy-upstream remote for pulls
-$ git remote add clippy-upstream https://github.com/rust-lang/rust-clippy
-# Make sure to not push to the upstream repo
-$ git remote set-url --push clippy-upstream DISABLED
-# Set a local remote
-$ git remote add clippy-local /path/to/rust-clippy
+[JOSH]: https://josh-project.github.io/josh/
+
+## Performing the sync from [`rust-lang/rust`] to Clippy
+
+Doing the sync now is just running
+
+```
+$ cargo dev sync pull
 ```
 
-> Note: The following sections assume that you have set those remotes with the
-> above remote names.
+This command will update the nightly toolchain in the `rust-toolchain` file and
+will pull the changes from the Rust repository.
 
-## Performing the sync from [`rust-lang/rust`] to Clippy
+If there should be merge conflicts, resolve them now and commit with the message
+`Merge from rustc`.[^1]
 
-Here is a TL;DR version of the sync process (all the following commands have
-to be run inside the `rust` directory):
-
-1. Clone the [`rust-lang/rust`] repository or make sure it is up-to-date.
-2. Checkout the commit from the latest available nightly. You can get it using
-   `rustup check`.
-3. Sync the changes to the rust-copy of Clippy to your Clippy fork:
-    ```bash
-    # Be sure to either use a net-new branch, e.g. `sync-from-rust`, or delete the branch beforehand
-    # because changes cannot be fast forwarded and you have to run this command again.
-    git subtree push -P src/tools/clippy clippy-local sync-from-rust
-    ```
-
-    > _Note:_ Most of the time you have to create a merge commit in the
-    > `rust-clippy` repo (this has to be done in the Clippy repo, not in the
-    > rust-copy of Clippy):
-    ```bash
-    git fetch upstream  # assuming upstream is the rust-lang/rust remote
-    git checkout sync-from-rust
-    git merge upstream/master --no-ff
-    ```
-    > Note: This is one of the few instances where a merge commit is allowed in
-    > a PR.
-4. Bump the nightly version in the Clippy repository by changing the date in the
-   rust-toolchain file to the current date and committing it with the message:
-   ```bash
-   git commit -m "Bump nightly version -> YYYY-MM-DD"
-   ```
-5. Open a PR to `rust-lang/rust-clippy` and wait for it to get merged (to
-   accelerate the process ping the `@rust-lang/clippy` team in your PR and/or
-   ask them in the [Zulip] stream.)
+> Note: If the version tests fail, refer to [bump version] in the release
+> documentation.
 
+Open a PR to `rust-lang/rust-clippy` and wait for it to get merged (to
+accelerate the process ping the `@rust-lang/clippy` team in your PR and/or ask
+them in the [Zulip] stream.)
```

Comment:
You usually just r+ these PRs directly. So is this just a safety net for new people doing the sync?

Reply:
This can be replaced to prefixed with the text from the other sync direction that you want to move.

> If you're a Clippy maintainer, you can `r+` the PR yourself. If not, change `r?
@ghost` to `r? clippy`.

Reply:
https://github.com/rust-lang/rust-clippy/pull/12762#discussion_r1638067919

I will move this line here.